From 0f1997e48fcceb5f741b9ff4d1312bceeda7d427 Mon Sep 17 00:00:00 2001
From: Ansuel Smith <ansuelsmth@gmail.com>
Date: Fri, 12 Nov 2021 00:30:45 +0100
Subject: [PATCH 2/3] dt-bindings: clock: qcom: Document use of qcom,nss-cc

Document use of qcom,nss-cc used to declare support for NSS Clock
Controller. NSS Core are Ubi32 core present on ipq806x SoC used as
Network Accellerator.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
---
 .../bindings/clock/qcom,nss-cc.yaml           | 96 +++++++++++++++++++
 1 file changed, 96 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/clock/qcom,nss-cc.yaml

diff --git a/Documentation/devicetree/bindings/clock/qcom,nss-cc.yaml b/Documentation/devicetree/bindings/clock/qcom,nss-cc.yaml
new file mode 100644
index 000000000000..12863a8dfb55
--- /dev/null
+++ b/Documentation/devicetree/bindings/clock/qcom,nss-cc.yaml
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/clock/qcom,nss-cc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: NSS Clock Controller
+
+maintainers:
+  - Ansuel Smith <ansuelsmth@gmail.com>
+
+description: |
+  NSS Network accellerator core present on ipq806x SoC. These additional
+  core must be scaled at the same time and require to be set to a safe
+  parent. Declare opp and supply to scale voltage with frequency.
+
+properties:
+  compatible:
+    const: qcom,nss-cc
+
+  clocks:
+    items:
+      - description: Ubi32 Core 1 source
+      - description: Ubi32 Core 2 source
+      - description: NSS tcm
+      - description: RPM NSS Fabric 0
+      - description: RPM NSS Fabric 1
+
+  clock-names:
+    items:
+      - const: ubi32-core1-clk-src
+      - const: ubi32-core2-clk-src
+      - const: nss-tcm-clk
+      - const: nss-fab0-clk
+      - const: nss-fab1-clk
+
+  nss-supply:
+    description:
+      Regulator supply for the NSS cores
+
+  operating-points-v2: true
+
+required:
+  - compatible
+  - clocks
+  - clock-names
+  - nss-supply
+  - operating-points-v2
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/qcom,gcc-ipq806x.h>
+    #include <dt-bindings/clock/qcom,rpmcc.h>
+
+    opp_table_nss: opp_table_nss {
+      compatible = "operating-points-v2";
+
+      opp-110000000 {
+        opp-hz = /bits/ 64 <110000000>;
+        opp-microvolt = <1100000>;
+        clock-latency-ns = <100000>;
+        opp-level = <0>;
+      };
+
+      opp-550000000 {
+        opp-hz = /bits/ 64 <550000000>;
+        opp-microvolt = <1100000>;
+        clock-latency-ns = <100000>;
+        opp-level = <1>;
+      };
+
+      opp-733000000 {
+        opp-hz = /bits/ 64 <733000000>;
+        opp-microvolt = <1150000>;
+        clock-latency-ns = <100000>;
+        opp-level = <2>;
+      };
+    };
+
+    nss-cc {
+      compatible = "qcom,nss-cc";
+      clocks = <&gcc UBI32_CORE1_CLK_SRC>,
+               <&gcc UBI32_CORE2_CLK_SRC>,
+               <&gcc NSSTCM_CLK>,
+               <&rpmcc RPM_NSS_FABRIC_0_CLK>,
+               <&rpmcc RPM_NSS_FABRIC_1_CLK>;
+      clock-names = "ubi32-core1-clk-src",
+                    "ubi32-core2-clk-src",
+                    "nss-tcm-clk",
+                    "nss-fab0-clk",
+                    "nss-fab1-clk";
+      operating-points-v2 = <&opp_table_nss>;
+      nss-supply = <&smb208_s1b>;
+    };
-- 
2.32.0

